workspace/src/string-transform.js

```diff
@@ -38,13 +38,207 @@ function transform_ins_ins(op0, op1) {
 function transform_del_ins(op0, op1) {
   return op1.pos <= op0.pos ? [op1] :
     op1.pos <= op0.pos + op0.length ? [makeOpIns(op0.pos, op1.value)] :
-    [makeOpIns(op1.pos - op0.length, op1.vlue)];
+    [makeOpIns(op1.pos - op0.length, op1.value)];
 }
 
 function transform_del_del(op0, op1) {
   return op1.pos <= op0.pos ? [op1] :
     op1.pos <= op0.pos + op0.length ? [makeOpDel(op0.pos, op1.value)] :
-    [makeOpDel(op1.pos - op0.length, op1.vlue)];
+    [makeOpDel(op1.pos - op0.length, op1.value)];
+}
+
+function transform_del_mov(op0, op1) {
+  var l1 = op1.length;
+  var l0 = op0.length;
+
+  // case 1: Mov area is left to Del area
+  /* Wrapped with block comment so the comment could be folded
+
+   // case 1.1:
+   //     ______________
+   //    |              |
+   //    |              v
+   //   [_____]     [--------]
+   //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+   //    op1           op0(del)
+   //
+   // case 1.2:
+   //     _______
+   //    |       |
+   //    |       v
+   //   [_____]     [--------]
+   //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+   //    op1           op0(del)
+   //
+   // case 1.3:
+   //     _______________________
+   //    |                       |
+   //    |                       v
+   //   [_____]     [--------]
+   //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+   //    op1           op0(del)
+   //
+   // case 1.4:
+   //  __
+   // |  |
+   // v  |
+   //   [_____]     [--------]
+   //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+   //    op1           op0(del)
+   */
+  // case 1
+  if (op1.pos + l1 <= op0.pos) {
+    if (op1.des >= op0.pos && op1.des <= op0.pos + l0) {
+      // case 1.1
+      return makeOpMov(op1.pos, l1, op0.pos);
+    } else if (op1.des >= op0.pos + l0) {
+      // case 1.3
+      return makeOpMov(op1.pos, l1, op0.op1.des - l0);
+    } else {
+      // case 1.2 & 1.4
+      return op1;
+    }
+  } else if (op1.pos <= op0.pos && op1.pos + l1 >= op0.pos && op1.pos + l1 <= op0.pos + l0) {
+    // case 2: Partial Overlap (Left to op0 area)
+    /*
+     // case 2.1:
+     //     ____________
+     //    |            |
+     //    |            V
+     //   {____[===}------]
+     //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+     //    op1       op0(del)
+
+     // case 2.2:
+     //     ___________________
+     //    |                   |
+     //    |                   V
+     //   {____[===}------]
+     //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+     //    op1       op0(del)
+
+     // case 2.3:
+     //  __
+     // |  |
+     // |  |
+     // v {____[===}------]
+     //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+     //    op1        op0(del)
+
+     */
+    if (op1.des >= op1.pos + l1 && op1.des <= op0.pos + l0) {
```

Comment:
Take a look at the transformIndex function. Can these logic be packaged and reused that function?